clldutils/text.py

```diff
@@ -0,0 +1,34 @@
+# coding: utf8
+from __future__ import unicode_literals, print_function, division
+import re
+
+def strip_brackets(text, brackets=None):
+    """Strip brackets and what is inside brackets from text.
+    
+    .. bite:: If the text contains only one opening bracket, the rest of the text
+      will be ignored. This is a feature, not a bug, as we want to avoid that
+      this function raises errors too easily.
+    """
+    brackets = brackets or {
+            "(": ")", "{": "}", "[": "]", "（": "）", "【": "】", "『": "』",
+            "«": "»", "⁽": "⁾", "₍": "₎"}
+    stack = []
+    tmp_sequence = list(text)
+    new_sequence = ''
+    while tmp_sequence:
```

Comment:
why not simply iterate:
```python
for itm in text:
    ...
```
At some point in my life - probably in python 2.5 times, though, iterating over characters in a string was noticeably faster than popping the first element in a loop.